Limit form button's width on large screens

zecakeh opened the pull request:
The button was too present at full width when using a large screen. I set a `minWidth` so it's noticeable enough even if the label is small.

![Screenshot_2021-01-26-10-47-31-797](https://user-images.githubusercontent.com/76261501/105829117-3d345280-5fc4-11eb-8be6-e827cfa0c2ab.jpeg)

Tested on web Firefox Linux
Tested on native Android

tasn commented:
It works, and it's great. But something about this change makes me feel uncomfortable. Maybe it's the increased presence of magic values in our codebase, maybe it's the use of align-self in a widget, I'm not sure. But I don't think there should be a widget that makes layouting decisions about itself.

I understand the importance of fixing the issue, I just really dislike the solution.

zecakeh commented:
- I agree with what you said. The breakpoints are taken from [Material Design](https://material.io/design/layout/responsive-layout-grid.html#breakpoints) (I know I mention Material Design a lot lately but I'm not that good at UI, except knowing when it looks wrong, so it helps to have a reference).
We could create a breakpoints enum in constants to make it clear what the breakpoints correspond to and have a clear justification for them.

- The `minWidth: 150` is arbitrary,  I chose something that looked good. Material Design uses a responsive layout grid but I thought it was too much to implement just for this.

- For me that's how flexbox works. If a child has a custom behavior, the behavior is set on that child. The only other option I see is nest the widget in a full width view, and the view decides if the widget is full width or aligned right, but I don't see the difference.

tasn commented:
>     * I agree with what you said. The breakpoints are taken from [Material Design](https://material.io/design/layout/responsive-layout-grid.html#breakpoints) (I know I mention Material Design a lot lately but I'm not that good at UI, except knowing when it looks wrong, so it helps to have a reference).
>       We could create a breakpoints enum in constants to make it clear what the breakpoints correspond to and have a clear justification for them.

That would be a good start, yeah. And just `useDisplayBreakpoint` that returns a value from this enum.

> 
>     * The `minWidth: 150` is arbitrary,  I chose something that looked good. Material Design uses a responsive layout grid but I thought it was too much to implement just for this.

Yeah, I don't think this is a good idea for a generic form button, there must be implications here that we are missing. Maybe setting padding could be better, but it still feels a bit off.

> 
>     * For me that's how flexbox works. If a child has a custom behavior, the behavior is set on that child. The only other option I see is nest the widget in a full width view, and the view decides if the widget is full width or aligned right, but I don't see the difference.

I understand that you can do it with flexbox. Though it feels very weird to control the layout in such an opaque way (doing it from inside a widget). It just feels wrong. Setting it externally feels more correct, but I guess that's just redundant. So maybe I should just accept it and not worry about it.

zecakeh commented:
> > * The `minWidth: 150` is arbitrary,  I chose something that looked good. Material Design uses a responsive layout grid but I thought it was too much to implement just for this.
> 
> Yeah, I don't think this is a good idea for a generic form button, there must be implications here that we are missing. Maybe setting padding could be better, but it still feels a bit off.

Yeah I don't think the padding is a good idea, because it wouldn't be necessary if the label was longer. It's just that the "Save" button looks really small compared to the rest.
Maybe setting the maxWidth of the form would make a small button look less weird?

tasn commented:
Could be. Looking at the page we are currently on: this is what Github seems to be doing.

zecakeh commented:
So I rebased and I think I solved all the problems mentioned:
- We use `useDeviceBreakpoint`
- I removed the `minWidth: 150` and I'll see in a future PR to limit the form's width
- No solution was provided for `alignSelf` so I left it
